File: apps/mobile/shared/src/commonMain/kotlin/xyz/scani/mobile/shared/data/WidgetData.kt
package xyz.scani.mobile.shared.data

import kotlinx.serialization.Serializable

@Serializable data class WidgetEntity(val id: String, val name: String, val value: String)

@Serializable
data class WidgetData(
    val portfolioTotal: String,
    val accounts: List<WidgetEntity>,
    val holdings: List<WidgetEntity>,
    val groups: List<WidgetEntity>,
    val vaults: List<WidgetEntity>,
    val updatedAt: Long,
)

File: apps/mobile/ios/ScaniWidgets/WidgetData.swift
import Foundation

struct WidgetEntity: Codable {
    let id: String
    let name: String
    let value: String
}

struct WidgetData: Codable {
    let portfolioTotal: String
    let accounts: [WidgetEntity]
    let holdings: [WidgetEntity]
    let groups: [WidgetEntity]
    let vaults: [WidgetEntity]
    let updatedAt: Int64
}

func loadWidgetData() -> WidgetData? {
    guard
        let url = FileManager.default
            .containerURL(forSecurityApplicationGroupIdentifier: "group.xyz.scani.mobile")?
            .appendingPathComponent("widget.json"),
        let data = try? Data(contentsOf: url)
    else { return nil }
    return try? JSONDecoder().decode(WidgetData.self, from: data)
}
